src/components/Button/Button.tsx

```diff
@@ -1,14 +1,46 @@
 import clsx from 'clsx';
-import React, { FC, HTMLAttributes, ReactChild } from 'react';
+import React, { ButtonHTMLAttributes, DetailedHTMLProps, FC } from 'react';
 
-export interface IButtonProps extends HTMLAttributes<HTMLButtonElement> {
-  children?: ReactChild;
+enum ButtonVariant {
+  PRIMARY = 'primary',
+  LINK = 'link',
 }
 
-export const Button: FC<IButtonProps> = (props) => {
-  const styles = clsx(
-    props.className,
-    'inline-flex items-center justify-center mt-3 px-4 py-2 w-full text-white font-medium bg-indigo-600 hover:bg-indigo-700 border border-transparent rounded-md focus:outline-none shadow-sm focus:ring-indigo-500 focus:ring-offset-2 focus:ring-2 sm:mt-0 sm:w-auto sm:text-sm',
-  );
-  return <button {...props} className={styles}></button>;
+enum ButtonSize {
+  XS = 'xs',
+  SM = 'sm',
+  MD = 'md',
+  LG = 'lg',
+  XL = 'xl',
+}
+
+export interface IButtonProps extends DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
+  variant?: `${ButtonVariant}`;
```

Comment:
Is this the same thing as `variant?: ButtonVariant;`?

Reply:
This is a template literal type, it'll enumerate the `enum` into a union of strings.  

https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html

Pretty useful, and it'll work with any iterable type I think

Reply:
To actually answer your question though, it's not the exactly the same, if you do just `ButtonVariant` it won't suggest a string, you would have to do `variant={ButtonVariant.Primary}`

Whereas this will generate a type that looks like: `primary | secondary | foo`
So you can then just do:

`variant="primary"` and it'll suggest it to you